src/main/java/org/springframework/data/web/config/PageableHandlerMethodArgumentResolverCustomizer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.web.config;
+
+import org.springframework.data.web.PageableHandlerMethodArgumentResolver;
+
+/**
+ * Callback interface that can be implemented by beans wishing to customize the
+ * {@link PageableHandlerMethodArgumentResolver} configuration.
+ *
+ * @author Vedran Pavic
+ */
+public interface PageableHandlerMethodArgumentResolverCustomizer {
```

Comment:
since this is going to be a feature for the `Kay` release we could add `@FunctionalInterface` here. Not sure if extending `Consumer<PageableHandlerMethodArgumentResolver>` and removing `customize` makes sense here.
Also I'd prefer a naming like `PageableHandlerMethodArgumentResolverConfigurationAdapter` over `...*Customizer`.